sql/2022/third-parties/distribution_of_lighthouse_unminified_css_by_3p.sql

```diff
@@ -0,0 +1,46 @@
+#standardSQL
+# Third-party pages with unminified CSS
+
+CREATE TEMPORARY FUNCTION getUnminifiedCssUrls(audit STRING)
+RETURNS ARRAY<STRUCT<url STRING, wastedBytes INT64>> LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(audit);
+  return $.details.items.map(({url, wastedBytes}) => {
+    return {url, wastedBytes};
+  });
+} catch (e) {
+  return [];
+}
+''';
+
+WITH base AS (
+  SELECT
+    page,
+    SUM(IF(is_3p, potential_savings, 0)) AS potential_savings
+  FROM (
+    SELECT
+      lighthouse.url AS page,
+      NET.HOST(data.url) IS NOT NULL AND NET.HOST(data.url) IN (
+        SELECT domain
+        FROM `httparchive.almanac.third_parties`
+        WHERE date = '2022-06-01' AND category != 'hosting'
+      ) AS is_3p,
```

Comment:
This kind of reads weird. I wonder if better done in a `WHERE` clause with a `LEFT OUTER JOIN` if you want to include the sites with non-3p savings? Took me a bit to get what you were doing here.

Reply:
I think this gives me the flexibility of changing the query to use the is_3p in the select. For example, on a separate ticket we are looking at including 1P too.

Then I can have,

```
  SUM(IF(is_3p, potential_savings, 0)) AS potential_3p_savings,
  SUM(potential_savings) AS potential_total_savings,
```

What do you think?

Reply:
Could do as a `LEFT OUTER JOIN` and then do:

```sql
  SUM(IF(third_parties.domain, potential_savings, 0)) AS potential_3p_savings,
  SUM(IF(third_parties.domain IS NULL, potential_savings, 0)) AS potential_1p_savings,
  SUM(potential_savings) AS potential_total_savings,
```

That would say you potentially having to repeat the subquery in both `potential_3p_savings` and `potential_1p_savings` columns.

Probably want to do a:
```sql
WITH third_parties AS (
  SELECT
    DISTINCT NET.HOST(domain)
  FROM
    `httparchive.almanac.third_parties`
)
```
 CTE in that case though to avoid joining multiple times if a domain is in there multiple times when NET.HOSTing it.

Or we could just go with that you have.

WDYT?

Reply:
Understood. Will see if I can improve it.

Reply:
Updated. Agreed on readability. Also updated some other similar queries to remove the nested `IN`.

The numbers varied very slightly. Any ideas?

If you're happy, it's good to go from my end.